### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by Jia-A**
This pull request contains homepage for my e-commerce website along with its footer, navbar, filter function and contexts.

### CODE REVIEW

**Review by shuklahimanshu81 — APPROVED** [review]
Just few suggestion : 
 1. Try to elaborate more in description with help of image or video.
 2. Try to open small PRs .
 Liked the way you break everything into functions. Everything looks good .

File: src/dot-components/Homepage/homepage.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  );
+};
+
+export { Homepage };
+
+{/* <div class="img-cont">
+    <img src="./images/paula-hawkins.jpg" class="round-img" alt="paula-hawkins-img"/>
+    <button class="small-text btn secondary-btn-border">Paula Hawkins</button>
+    </div>
+*/}
```
Comment (shuklahimanshu81) [inline src/dot-components/Homepage/homepage.jsx]: Remove these comments

**Review by shuklahimanshu81 — APPROVED** [review]
Everything else looks good to me . Good for merge.

File: src/Context/filterContext.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    createContext,
+    useContext,
+    useReducer, 
+    useState, 
+    useEffect
+} from "react";
+
+import axios from "axios";
+
+
```
Comment (Megha-Joshi) [inline src/Context/filterContext.js]: Remove extra spacing.

File: src/Context/filterContext.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+const useFilter = () => useContext(FilterContext);
+
+export {
+    useFilter,
+    FilterProvider
+};
+
+
+
```
Comment (Megha-Joshi) [inline src/Context/filterContext.js]: Remove extra spacing

File: src/dot-components/Homepage/homepage.css [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    display: flex;
+    flex-direction: row;
+    gap: 2.5rem;
+    align-items: center;
+    justify-content: center;
+    padding: 1.875rem;
+    border-radius: var(--border-radius);
+    flex-wrap: wrap;
+    margin-top: 10px;
+
```
Comment (saadahmadhere) [inline src/dot-components/Homepage/homepage.css]: remove extra spaces

File: src/Context/wishlistContext.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                ...wishState,
+                wish: [
+                    ...wishState.wish,
+                    {...action.payload},
+            ],}
+        default:
+            return wishState;
+    
+            
+    }
```
Comment (Megha-Joshi) [inline src/Context/wishlistContext.js]: Remove extra spacing.

File: src/dot-components/Footer/footer.css [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    align-items: center;
+    box-shadow: 2px 2px 6px 0px rgba(0, 0, 0, 0.1);
+    background-color: #0c172e;
+    padding: 10px;
+
+}
+.fab,
+.quote, 
+.copyright{
+    color: var(--whitesmoke-color)
```
Comment (Megha-Joshi) [inline src/dot-components/Footer/footer.css]: Nice use of CSS variables.

File: src/dot-components/Homepage/homepage.css [hunk 7]
```diff
@@ -0,0 +1,87 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,300;0,400;0,500;0,900;1,300&display=swap');
```
Comment (Megha-Joshi) [inline src/dot-components/Homepage/homepage.css]: Include this in root.css

**Review by saadahmadhere — COMMENTED** [review]
Just take care of the extra spacing, using an extension like prettier would work fine. Rest is well and good.

File: src/dot-components/Homepage/homepage.css [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    color: var(--font-color);
+}
+.read-btn{
+    padding: 0px;
+    margin: 15px 0px 0px 0px;
+    background-color: transparent;
+}
+.complete-card {
+    max-width: 40rem;
+    background-color: #E1F0C4;
```
Comment (Megha-Joshi) [inline src/dot-components/Homepage/homepage.css]: Use CSS variable for giving color.

File: src/dot-components/Homepage/homepage.jsx (hunk 1 above)
Comment (Megha-Joshi) [reply src/dot-components/Homepage/homepage.jsx]: +1

File: src/public-css/sign-login-auth.css [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    padding: 20px;
+
+
+    text-align: center;
+    padding: 1.25rem;
+
+    align-items: center;
+    justify-content: center;
+    text-align: center;
+    padding: 20px;
```
Comment (Megha-Joshi) [inline src/public-css/sign-login-auth.css]: As this value is big, you can use rem instead of px.

**Review by Megha-Joshi — APPROVED** [review]
- Good naming conventions.
- Good folder structure.
- Suggestion: Try to give screenshot or video of your UI design to get better review.
- Just do the suggested changes otherwise good to merge.

**Pull request merged**